webcomponents/tool/src/main/frontend/package.json

```diff
@@ -5,7 +5,7 @@
   "main": "js/sakai-element.js",
   "workspaces": [ "./packages/*" ],
   "scripts": {
-    "lint": "eslint --fix \"packages/**/src/*.js\"",
```

Comment:
Removing the `--fix` option might be debatable. But I would be in favor of removing it, like you did it here. I wouldn't want the source code to change when I run a build command.

Reply:
I agree: a build should not modify the code.

I am okay with a git commit enforcing a specific style.

I'm also okay with blocking a PR if it doesn't comply with eslint rules

Reply:
I don't think people are going to enjoy going in and fixing eslint's errors manually. I think that if we remove the --fix option then people will start just weakening the eslint rules to make their lives easier. Then there will be no point in linting at all. That's my opinion. I think we should ask on the dev list - I'll email the list now.

Reply:
I see that running `npm run lintfix` after a broken build might be inconvenient. Personally I wouldn't mind it, because then I am aware of the changes. But I am not touching webcomponents often, so I am fine with whatever others decide.

Reply:
Will the pre commit hook still work? The husky stuff? As long as that happens, then badly formatted stuff will not even make it to the github build stage.

Reply:
the linting is not just about formatting though. It catches a lot of sources of js bugs, like variable uses without a declaration, the use of var instead of constants. If we want our JS to be reliable, without taking the horrible leap to typescript, then we really need to lint and fix. Anyway, the international Sakai jury can decide.

Reply:
I think you're the only person here who knows what the husky stuff is! 

But yes, a git hook that auto-lints on commit makes perfect sense to me. Then it's more or less impossible to submit non-linted code

Reply:
Husky adds a pre commit hook to lint and analyze your staged changes. You can test it by adding some crap to one of the webcomponent files and trying to commit. Try adding:
var banana = "eggs";

to a constructor and then commit. You should see the hook running through the lint and analyze stages.

Reply:
why do you need a jury when we seem to more or less agree?

we want linting. we don't want any bad commits making it in. we like lint rules. 

but a build seems like an odd time to do the *fixing*....

Reply:
Not being snarky. I'm just okay with a group of people deciding.

Reply:
the outstanding question for me is: Does the pre-commit hook work on Windows?

Reply:
i don't see any husky running for me locally. are you sure it's installed correctly?

Reply:
Hmmm. That's a worry. I'll have a dig into that.

Reply:
pre-commit concept should work on all platforms.

eslint continues to run on all builds (including CI) and will stop the build if it fails. there is no way to merge a PR into sakaiproject/sakai that breaks eslint rules

pre-commit is definitely the right place to format files, auto-fix files, etc. but i don't see any pre-commit hooks defined in .git/hooks/

Reply:
Husky installs them under frontend/.husky in webcomponents. I'm just doing a bit of testing.

Reply:
yeah i see it there. but i guess i was expecting a .git/hooks/pre-commit that would enforce it?  do you have something there? 

and what about a "prepare" section in the package.json file?

Reply:
Add this to package.json

"prepare": "cd ../../../../../ && husky install webcomponents/tool/src/main/frontend/.husky"

Reply:
Also, pre-commit in .husky needs to be executable

https://stackoverflow.com/questions/68367259/husky-needs-to-make-executable-for-every-new-branch

Reply:
ok let's move to a separate PR

Reply:
I have it working locally, but using chmod to update the executable bit on the pre-commit file. So, not windows friendly. However, I don't think the executable thing is an issue on windows?

Reply:
seems working here. i didn't need to do a chmod

Reply:
It seems this was included under another piece of code (push notifications), and it's not working on Windows